niflow_manager/cli/build.py

```diff
@@ -0,0 +1,190 @@
+        if params_entry:
+            post_build["entrypoint"] = params_entry
+        else:
+            post_build[
+                "entrypoint"
+            ] = f"/opt/miniconda-latest/envs/testkraken/bin/{workflow_path.name}"
+
+    for key, spec in post_build.items():
+        if key == "miniconda":
+            if miniconda_env:
+                miniconda_dict = {"use_env": miniconda_env}
+            else:
+                miniconda_dict = {"create_env": "testkraken"}
+            miniconda_dict.update(spec)
+            instructions.append(("miniconda", miniconda_dict))
+        else:
+            instructions.append((key, spec))
+
+    return {"pkg_manager": pkg_manager, "instructions": tuple(instructions)}
+
+
+def write_dockerfile_sp(nrd_jsonfile, dockerfile):
+    """ Generate and write Dockerfile to `dockerfile`, uses Neurodocker cli
+        These is a tmp function, would prefer to use write_dockerfile
+    """
+    nrd_args = [
+        "neurodocker",
+        "generate",
+        "docker",
+        nrd_jsonfile,
+        "-o",
+        dockerfile,
+        "--no-print",
+        "--json",
+    ]
+    # not sure if I need to use out_json anywhere, might remove "--json"
+    out_json = sp.run(nrd_args, check=True, stdout=sp.PIPE).stdout.decode()
+
+
+def build_image(dockerfile, workflow_path, tag=None, build_opts=None):
+    """Build Docker image. TODO
+
+    Parameters
+    ----------
+    dockerfile : path-like
+        Path to Dockerfile. May be absolute or relative. If `build_context`
+        if provided, `dockerfile` is joined to `build_context`.
+    workflow_path : path-like
+        Path to workflow that will be used as a build context.
+    tag : str
+        Docker image tag. E.g., "kaczmarj/myimage:v0.1.0".
+    build_opts : str
+        String of options to pass to `docker build`.
+    """
+    tag = "" if tag is None else "-t {}".format(tag)
+    build_opts = "" if build_opts is None else build_opts
+
+    cmd_base = "docker build {tag} {build_opts}"
+    cmd = cmd_base.format(tag=tag, build_opts=build_opts)
+
+    build_context = workflow_path
+    # changing build directory, needed for fnp
+    # was failing wit providing build_context to build command)
+    cwd = os.getcwd()
+    os.chdir(build_context)
+    cmd += " -f {} .".format(dockerfile)
+
+    sp.run(cmd.split(), check=True)
+    os.chdir(cwd)
+
+
+def docker_image(workflow_path, working_dir=None):
+    """the main function to create Dockerfile and build the image"""
+    workflow_path = Path(workflow_path).absolute()
+    if working_dir:
+        working_dir = Path(working_dir).absolute()
+    else:
+        working_dir = Path(
+            tempfile.mkdtemp(prefix=f"nfm-{workflow_path.name}")
+        ).absolute()
+    dockerfile = working_dir / "Dockerfile"
+    if dockerfile.exists():
+        dockerfile.unlink()
+    jsonfile = working_dir / f"nrd_spec.json"
+    if jsonfile.exists():
+        jsonfile.exists()
```

Comment:
this is weird.

Reply:
or very "philosophical" - `if True: True` ;-)

I believe it was meant to be `jsonfile.unlink()`